### PR TITLE
Feature/clean up liteserv pt2

### LINKED ISSUE #722: Refactor LiteServ.py
Getting too large, time for a redesign.
- [ ] Rename net -> dotnet

### LINKED ISSUE #746: Disable longpoll timeout tests macosx

### LINKED ISSUE #757: Reenable large inline attachments client sg tests
When this is fixed - https://github.com/couchbase/couchbase-lite-net/issues/749

### PR DESCRIPTION
**Pull request by sethrosetter**
- [x] Expose platform versions to unit tests
- [x] Run client P2P tests
- [x] Run functional tests against branch
- [x] Add Jenkins job for framework unit tests (In Progress)

Fixes #722 
Fixes #757 
Fixes #746 

Changes
- Refactor LiteServ.py into platform specific subclasses with a shared base class.
- Disable large inline attachments test
- Skip long poll timeout tests for mac osx rather than fail

Passing unit tests: http://uberjenkins.sc.couchbase.com/job/mobile-testkit-framework-tests/9/

### CODE REVIEW

**Comment by EricACooper** [comment]
Looks good - thanks for taking the time to improve the code.

File: keywords/LiteServBase.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+from requests.sessions import Session
+from requests.exceptions import ConnectionError
+
+from keywords.constants import MAX_RETRIES
+from keywords.exceptions import LiteServError
+from keywords.utils import log_r
+from keywords.utils import log_info
+
+
+class LiteServBase(object):
```
Comment (sethrosetter) [inline keywords/LiteServBase.py]: Done

File: framework_tests/test_liteserv_android.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+import datetime
+
+from keywords.constants import RESULTS_DIR
+from keywords.constants import BINARY_DIR
+from keywords.LiteServFactory import LiteServFactory
+from keywords.MobileRestClient import MobileRestClient
+
+
+@pytest.fixture(scope="function")
+def setup_liteserv_android_sqlite(request):
```
Comment (sethrosetter) [inline framework_tests/test_liteserv_android.py]: Done

File: framework_tests/test_liteserv.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+def test_running_liteserv(request):
+    liteserv = LiteServFactory.create("macosx",
+                                      version_build="1.3.1-6",
+                                      host="localhost",
+                                      port=59840,
+                                      storage_engine="SQLite")
+
+    liteserv.download()
```
Comment (sethrosetter) [inline framework_tests/test_liteserv.py]: I think it is ok if we download. I'm not seeing another way at the moment.

File: framework_tests/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+Running the framework unit tests
+
```
Comment (sethrosetter) [inline framework_tests/README.md]: Done

**Comment by sethrosetter** [comment]
- Removed all test code duplication
- Skipping test marked flakey (windows ansible). This is a ticket to reenable when we update to latest ansible.  https://github.com/couchbaselabs/mobile-testkit/issues/774

**Comment by sethrosetter** [comment]
Passing jenkins job - http://uberjenkins.sc.couchbase.com/job/mobile-testkit-tests/

**Pull request merged**